app/src/main/java/com/celzero/bravedns/ui/HomeScreenActivity.kt

```diff
@@ -132,6 +137,41 @@ class HomeScreenActivity : AppCompatActivity(R.layout.activity_home_screen) {
         insertDefaultData()
 
         initUpdateCheck()
+
+        detectAppExitInfo()
+    }
+
+    private fun detectAppExitInfo() {
+        if (!isAtleastR()) return
+
+        if (am == null) {
+            am = getSystemService(ACTIVITY_SERVICE) as ActivityManager
```

Comment:
Move this initialzation to `init` block, that should get rid of runtime null pointer asserts, like in line 154.

Reply:
this code block does not exist. refactored